authprox/service.go

```diff
@@ -0,0 +1,354 @@
+package authprox
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/binary"
+	"errors"
+	"fmt"
+
+	bolt "github.com/coreos/bbolt"
+	"github.com/dedis/cothority"
+	"github.com/dedis/kyber"
+	"github.com/dedis/kyber/share"
+	"github.com/dedis/kyber/sign/dss"
+	"github.com/dedis/kyber/suites"
+	"github.com/dedis/onet"
+	"github.com/dedis/onet/log"
+	"github.com/dedis/onet/network"
+	"github.com/dedis/protobuf"
+)
+
+// ServiceName is the name of the Authentication Proxy service.
+const ServiceName = "AuthProx"
+
+var authProxID onet.ServiceID
+
+type service struct {
+	*onet.ServiceProcessor
+	ctx        context.Context
+	validators map[string]Validator
+	db         *bolt.DB
+	bucket     []byte
+}
+
+func init() {
+	var err error
+	authProxID, err = onet.RegisterNewService(ServiceName, newService)
+	if err != nil {
+		log.ErrFatal(err, "could not register")
+	}
+	network.RegisterMessages(
+		&EnrollRequest{}, &EnrollResponse{},
+		&SignatureRequest{}, &SignatureResponse{},
+		&EnrollmentsRequest{}, &EnrollmentsResponse{}, &EnrollmentInfo{},
+		&ti{}, &dssConfig{},
+	)
+}
+
+// A Validator is able to check the provided authInfo with respect to
```

Comment:
Another detail (nothing needs to be changed): I like starting service-files with constants, `init`, structures, `NewStruct`, public methods, then private methods and the `newService` at the bottom. This helps me navigate the file.